core/domain/user_query_jobs_one_off.py

```diff
@@ -105,6 +105,11 @@ def map(user_settings_model):
         if not query_criteria_satisfied:
             return
 
+        email_preferences = user_services.get_email_preferences(user_id)
+
+        if not email_preferences.can_receive_email_updates:
+            return
```

Comment:
Suggest filtering this at the beginning since it should be a quick short-circuit (e.g. don't process any data if we don't intend to use it).

Reply:
+1, agreed